l2-contracts/src/L2RewardManagerStorage.sol

```diff
@@ -1,14 +1,47 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity >=0.8.0 <0.9.0;
 
-import { IL2RewardManager } from "./interface/IL2RewardManager.sol";
-
 /**
  * @title L2RewardManagerStorage
  * @author Puffer Finance
  * @custom:security-contact security@puffer.fi
  */
 abstract contract L2RewardManagerStorage {
+    /**
+     * @notice A record of a single epoch for storing the rate and root.
+     * @param startEpoch The start epoch of the interval
```

Comment:
I do not understand our usage of `startEpoch` and `endEpoch`. Each claiming interval is specific to a merkle tree but the code requires users to enter their `startEpoch` and `endEpoch` to reconstruct their merkle leaf. This costs more gas, adds room for error and complexity to track off-chain, and ultimately is just used to call `getIntervalID()`. Can we simplify these contracts to just deal with an input `intervalID` and just use a counter. For example, the first claiming interval is just `bytes32(0)`, the next is `bytes32(1)`, and so on.

Reply:
startEpoch and endEpoch are the ID of the claiming interval. IMO having a counter would cause confusion, because then,. we would have to connect the claiming interval to some N idx.

In the contracts, we can accept bytes32 claimingId, and proof. And add indexed to claimingId events.